### PR TITLE
dwallet-network/eth-light-client: add cli command `connect-eth-dwallet`

### PR DESCRIPTION
**Pull request by yuvalelmo**

### CODE REVIEW

File: crates/sui-types/src/eth_dwallet.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+/// Rust version of the Move dwallet_system::ethereum_state::LatestEthereumState type
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct LatestEthStateObject {
+    pub id: UID,
+    pub eth_state_id: ObjectID,
+    pub time_slot: u64,
+}
+
+/// Rust version of the Move dwallet_system::ethereum_state::EthState type
```
Comment (zeevmoney) [inline crates/sui-types/src/eth_dwallet.rs:36]:
```suggestion
/// Rust version of the Move [`dwallet_system::ethereum_state::EthState`] type.
```

File: crates/sui/src/client_commands.rs [hunk 2]
```diff
@@ -89,6 +89,7 @@ use tokio::time::sleep;
 use tracing::info;
 
 use crate::ethereum_client_commands::EthClientCommands;
+use crate::ethereum_client_commands::{create_eth_dwallet, EthClientCommands};
```
Comment (zeevmoney) [inline crates/sui/src/client_commands.rs:92]: Note that `EthClientCommands` was imported twice (see line 91).

File: crates/sui/src/ethereum_client_commands.rs [hunk 3]
```diff
@@ -1,5 +1,19 @@
+use anyhow::anyhow;
 use clap::Subcommand;
+use serde::{Deserialize, Serialize};
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:3]:
```suggestion
```

File: crates/sui/src/ethereum_client_commands.rs [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+use serde_json::{Number, Value};
+use shared_crypto::intent::Intent;
+use sui_json::SuiJsonValue;
+use sui_json_rpc_types::{SuiExecutionStatus, SuiTransactionBlockEffectsAPI};
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:7]:
```suggestion
use sui_json_rpc_types::SuiExecutionStatus;
```

File: crates/sui/src/ethereum_client_commands.rs [hunk 5]
```diff
@@ -87,3 +101,51 @@ pub enum EthClientCommands {
...
+    )
+    .await?;
+    Ok(serialize_or_execute!(
+        tx_data,
+        serialize_unsigned_transaction,
+        serialize_signed_transaction,
+        context,
+        Call
+    ))
+}
```
Comment (zeevmoney) [inline crates/sui/src/ethereum_client_commands.rs:151]: Fix all `unwrap()`.

**Review by zeevmoney — CHANGES_REQUESTED** [review]
see comments

**Review by zeevmoney — APPROVED** [review]

File: crates/sui-types/src/eth_dwallet.rs [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+pub const CREATE_ETH_DWALLET_CAP_FUNC_NAME: &IdentStr = ident_str!("create_eth_dwallet_cap");
+pub const VERIFY_ETH_STATE_FUNC_NAME: &IdentStr = ident_str!("verify_new_state");
+pub const APPROVE_MESSAGE_FUNC_NAME: &IdentStr = ident_str!("approve_message");
+
+/// Rust version of the Move [`dwallet_system::eth_dwallet::EthDWalletCap`] type.
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, Eq, PartialEq)]
+pub struct EthDWalletCap {
+    pub id: UID,
+    pub dwallet_cap_id: ID,
+    pub eth_smart_contract_addr: String,
```
Comment (ycscaly) [inline crates/sui-types/src/eth_dwallet.rs:26]:
```suggestion
    pub eth_smart_contract_address: String,
```

Please use fully abbreviated names

File: crates/sui-types/src/eth_dwallet.rs [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+pub const LATEST_ETH_STATE_STRUCT_NAME: &IdentStr = ident_str!("LatestEthereumState");
+pub const ETH_STATE_STRUCT_NAME: &IdentStr = ident_str!("EthState");
+pub const INIT_STATE_FUNC_NAME: &IdentStr = ident_str!("init_state");
+pub const CREATE_ETH_DWALLET_CAP_FUNC_NAME: &IdentStr = ident_str!("create_eth_dwallet_cap");
+pub const VERIFY_ETH_STATE_FUNC_NAME: &IdentStr = ident_str!("verify_new_state");
+pub const APPROVE_MESSAGE_FUNC_NAME: &IdentStr = ident_str!("approve_message");
+
+/// Rust version of the Move [`dwallet_system::eth_dwallet::EthDWalletCap`] type.
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, Eq, PartialEq)]
+pub struct EthDWalletCap {
```
Comment (ycscaly) [inline crates/sui-types/src/eth_dwallet.rs:23]:
```suggestion
pub struct EthereumDWalletCap {
```
Please use fully abbreviated names

**Review by ycscaly — APPROVED** [review]

**Review by zeevmoney — APPROVED** [review]

**Pull request merged**